CONTRIBUTING.md

```diff
@@ -0,0 +1,292 @@
+
+## Editor Configurations
+
+Configure your editor to use our lint and code style rules to speed up the code review process!
+
+### Code Style
+
+UTAM Java uses the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).
+Format settings definition files for importing into IDEs are available for
+[Eclipse](https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml)
+and [IntelliJ IDEA](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
+
+## Git Workflow
+
+The process of submitting a pull request is straightforward and generally follows the same pattern each time:
+
+1. [Fork the UTAM Java repo](#fork-the-utam-java-repo)
+2. [Create a feature branch](#create-a-feature-branch)
+3. [Make your changes](#make-your-changes)
+4. [Rebase](#rebase)
+5. [Check your submission](#check-your-submission)
+6. [Create a pull request](#create-a-pull-request)
+7. [Update the pull request](#update-the-pull-request)
+8. [Commit Message Guidelines](#commit-message-conventions)
+
+### Fork the UTAM Java repo
+
+[Fork][fork-a-repo] the [salesforce/utam-java](https://github.com/salesforce/utam-java) repo. Clone your fork in your local
+workspace and [configure][configuring-a-remote-for-a-fork] your remote repository settings.
+
+```bash
+$ git clone git@github.com:<YOUR-USERNAME>/utam-java.git
+$ cd utam-java
+$ git remote add upstream git@github.com:salesforce/utam-java.git
+```
+
+### Create a feature branch
+
+```bash
+$ git checkout main
+$ git pull origin main
+$ git checkout -b <name-of-the-feature>
+```
+
+### Make your changes
+
+Modify the files, lint, format and commit your code using the following commands:
+
+```bash
+$ git add <path/to/file/to/commit>
+$ git commit
+$ git push origin <username>/<name-of-the-feature>
+```
+
+Commit your changes using a descriptive commit message that follows our [Commit Message Guidelines](#commit-message-conventions). Adherence
+to these conventions is necessary because release notes will be automatically generated from these messages. NOTE:
+optional use of _git cz_ command triggers interactive semantic commit, which prompts user with commit related questions,
+such as commit type, scope, description, and breaking changes. Use of _git cz_ is optional but recommended to ensure
+format consistency.
+
+The above commands will commit the files into your feature branch. You can keep pushing new changes into the same branch
+until you are ready to create a pull request.
+
+### Check your submission
+
+#### Test and lint your changes
+Test your changes using the unit tests by running the following command:
+
+```bash
+mvn test
+```
+
+Note that a correct submission should contain passing unit tests for the modified code.
+
+### Rebase
+
+Sometimes your feature branch will get stale with respect to the master branch, and it will require a rebase. The
+following steps can help:
+
+```bash
+$ git checkout main
```

Comment:
git fetch upstream should be first here, since we're assuming user works from fork

Reply:
fixed